rust/src/thermal.rs

```diff
@@ -169,17 +183,37 @@ struct FanStateLevels {
 #[derive(Default)]
 struct FanState {
     rpm: f64,
+    rpm_success: bool,
     rpm_bounds: FanRpmBounds,
+    bounds_success: bool,
     state_levels: FanStateLevels,
+    levels_success: bool,
     samples: SampleBuf<u32, MAX_SAMPLES>,
 }
 
 impl FanState {
     fn update(&mut self) {
-        self.rpm = get_fan_rpm();
-        self.rpm_bounds = get_fan_bounds();
-        self.state_levels = get_fan_levels();
-        self.samples.insert(self.rpm as u32);
+        if let Ok(rpm) = get_fan_rpm() {
+            self.rpm = rpm;
+            self.samples.insert(rpm as u32);
+            self.rpm_success = true;
+        } else {
+            self.rpm_success = false;
+        }
+
+        if let Ok(rpm_bounds) = get_fan_bounds() {
+            self.rpm_bounds = rpm_bounds;
+            self.bounds_success = true;
+        } else {
+            self.bounds_success = false;
```

Comment:
it looks like bounds_success isn't actually consumed anywhere? is that intentional?